test/cases/compile_errors/use_implicit_casts_to_assign_null_to_non-nullable_pointer.zig

```diff
@@ -12,4 +12,4 @@ export fn entry() void {
 //
 // :4:24: error: expected type '*?*i32', found '**i32'
 // :4:24: note: pointer type child '*i32' cannot cast into pointer type child '?*i32'
-// :4:24: note: mutable '*i32' allows illegal null values stored to type '?*i32'
+// :4:24: note: mutable '?*i32' would allow illegal null values stored to type '*i32'
```

Comment:
sorry, this is an extreme nitpick but is the word "would" accomplishing anything here? The programmer already knows that the action did not occur since there is a compile error instead of a succeeded compilation

Reply:
I'm happy to nitpick this one a little, this error message is quite awkward to phrase understandably (as evidenced by these types having been printed backwards for god knows how long).

I think this phrasing is clearer, because it makes the relationship between the two types here more obvious: in this case, I already *have* a pointer to `*i32`, and I'm *trying* to coerce it to a pointer to `?*i32`, but if I got that value, it *would* allow an illegal operation. If the phrasing is "allows", then it could be interpreted as "I already have a mutable pointer to `?*i32`, and I'm trying to coerce to a pointer to `*i32`, but my existing pointer allows illegal nulls", which is a different thing.